Inventory.cs

```diff
@@ -0,0 +1,82 @@
+﻿public class Inventory
+{
+    private List<Product> products = new();
+
+    public void Add(string name, decimal price, int quantity)
+    {
+        var newProduct = new Product(name, price, quantity);
+        if (Product.AreValid(price, quantity)) 
+        {
+            products.Add(newProduct);
+        }
+    }
+
+    public bool IsEmpty()
+    {
+        return products.Count == 0;
+    }
+
+    public void Print(int index = -1)
+    {
+        if (IsEmpty())
+            Console.WriteLine("There are no products.");
+
+        // Print a specific product
+        else if (index != -1)
+        {
+            Console.WriteLine($"Name: {products[index].Name}");
+            Console.WriteLine($"Price: {products[index].Price}");
+            Console.WriteLine($"Quantity: {products[index].Quantity}");
+        }
+
+        // Print all of the existing products (without passing an argument)
+        else
+        {
+            for (var i = 0; i < products.Count; i++)
+            {
+                Console.Write($"Product #{i + 1}:\n");
+                Console.WriteLine($"\tName: {products[i].Name}");
+                Console.WriteLine($"\tPrice: {products[i].Price}");
+                Console.WriteLine($"\tQuantity: {products[i].Quantity}\n");
+            }
+        }
+    }
+
+    public void Edit(int index, object newValue)
```

Comment:
rename to **EditProduct**

Reply:
that's a very interesting way to update an object property :D  No need to use generics, What if I asked you now to add another property to product to store the product SKU, which is also a string, then this method won't work out. You can just have a separate method for each update.